pkg/kubectl/rolling_updater.go

```diff
@@ -411,13 +417,14 @@ func (r *RollingUpdater) readyPods(oldRc, newRc *api.ReplicationController) (int
 			return 0, 0, err
 		}
 		for _, pod := range pods.Items {
-			if api.IsPodReady(&pod) {
-				switch controller.Name {
-				case oldRc.Name:
-					oldReady++
-				case newRc.Name:
-					newReady++
-				}
+			if r.nowFn != nil && !deployment.IsPodAvailable(&pod, minReadySeconds, r.nowFn().Time) {
```

Comment:
If nowFn is nil default to Now(). Otherwise, you will consider every pod as ready, right?
```go
if r.nowFn == nil {
    r.nowFn = unversioned.Now()
}
if !deployment.IsPodAvailable(&pod, minReadySeconds, r.nowFn().Time) {
    continue
}
```